File: app/src/main/java/com/jmdevelopers/workproject/Model/Doavel.java
package com.jmdevelopers.workproject.Model;

import com.google.firebase.database.DatabaseReference;
import com.jmdevelopers.workproject.Config.ConfiguracaoFirebase;

public class Doavel {
    private String nome;
    private String foto;
    private String foto2;

    private String descricao;
    private String validade;
    private String telefone;
    private String tipo;
    private String dataPublicada;
    private String Localizacao;
    public void salvar(){

        DatabaseReference firebase = ConfiguracaoFirebase.getFirebaseDatabase();
        firebase.child("doacao")
                .push()
                .setValue(this);

    }
    public String getFoto2() {
        return foto2;
    }

    public void setFoto2(String foto2) {
        this.foto2 = foto2;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getFoto() {
        return foto;
    }

    public void setFoto(String foto) {
        this.foto = foto;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public String getValidade() {
        return validade;
    }

    public void setValidade(String validade) {
        this.validade = validade;
    }

    public String getTelefone() {
        return telefone;
    }

    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getDataPublicada() {
        return dataPublicada;
    }

    public void setDataPublicada(String dataPublicada) {
        this.dataPublicada = dataPublicada;
    }

    public String getLocalizacao() {
        return Localizacao;
    }

    public void setLocalizacao(String localizacao) {
        Localizacao = localizacao;
    }

    public Doavel() {
    }
}

File: app/src/main/java/com/jmdevelopers/workproject/model/Doavel.kt
package com.jmdevelopers.workproject.model

import com.jmdevelopers.workproject.services.DoacaoService
import java.util.*

class Doavel() {

    var nome: String? = null
    var descricao: String? = null
    var validade: String? = null
    var telefone: String? = null
    var tipo: String? = null
    var dataPublicada: String? = null
    var localizacao: String? = null
    val fotos: ArrayList<String>? = arrayListOf()

    constructor(nome: String, descricao: String, validade: String, telefone: String, tipo: String, localizacao: String) : this() {
        this.nome = nome
        this.descricao = descricao
        this.validade = validade
        this.telefone = telefone
        this.tipo = tipo
        this.localizacao = localizacao
    }

    fun salvar() {
        DoacaoService().api.push().setValue(this)
    }
}
